why is the api not stable in minor releases?

### Initial checklist

- [X] I understand this is a bug report and questions should be posted in the [Community Forum](https://community.transloadit.com/)
- [X] I searched [issues](https://github.com/transloadit/uppy/issues?q=is%3Aissue) and couldn’t find anything (or linked relevant results below)

### Link to runnable example

uppy core

### Steps to reproduce

this commit:

https://github.com/transloadit/uppy/pull/4128

made the cancel and other methods suddenly have a mandatory parameter

That by itself is fine, but then there is a release done i guess 3.0.3 so really a minor release that now breaks all our builds..

We have a system that we do update minor releases because of the security fixes and so on, but we don't expect then suddenly api breaking changes of existing api..

These kind of changes should be in a 4.x release or at least a 3.x release.. not just a 3.0.x 

Are those new params mandatory? else the solution would have been to just make then optional, now the typescript compiler just breaks:


```
     [java] Error: packages/@servoy/servoyextracomponents/src/multifileupload/multifileupload.ts:230:19 - error TS2554: Expected 1 arguments, but got 0.
     [java] 
     [java] 230         this.uppy.cancelAll();
     [java] ~~~~~~~~~~~
     [java] 
     [java] node_modules/@uppy/core/types/index.d.ts:359:13
     [java] 359   cancelAll(options: CancelOptions): void
     [java] ~~~~~~~~~~~~~~~~~~~~~~
     [java] An argument for 'options' was not provided.
     [java] 
     [java] 
     [java] Error: packages/@servoy/servoyextracomponents/src/multifileupload/multifileupload.ts:242:19 - error TS2554: Expected 1 arguments, but got 0.
     [java] 
     [java] 242         this.uppy.cancelAll();
     [java] ~~~~~~~~~~~
     [java] 
     [java] node_modules/@uppy/core/types/index.d.ts:359:13
     [java] 359   cancelAll(options: CancelOptions): void
     [java] ~~~~~~~~~~~~~~~~~~~~~~
     [java] An argument for 'options' was not provided.
     [java] 
     [java] 
     [java] Error: packages/@servoy/servoyextracomponents/src/multifileupload/multifileupload.ts:258:19 - error TS2554: Expected 1 arguments, but got 0.
     [java] 
     [java] 258         this.uppy.close();
     [java] ~~~~~~~
     [java] 
     [java] node_modules/@uppy/core/types/index.d.ts:378:9
     [java] 378   close(options: CancelOptions): void
     [java] ~~~~~~~~~~~~~~~~~~~~~~
     [java] An argument for 'options' was not provided.
```

### Expected behavior

that we don't have breaking api (that typescript compiler now chokes at) in minor releases.


### Actual behavior

we now get "compile" errors

--- a/packages/@uppy/core/types/index.d.ts
+++ b/packages/@uppy/core/types/index.d.ts
@@ -356,7 +356,7 @@ export class Uppy {
     UploadResult<TMeta>
   >
 
-  cancelAll(options: CancelOptions): void
+  cancelAll(options?: CancelOptions): void
 
   retryUpload<TMeta extends IndexedObject<any> = Record<string, unknown>>(
     fileID: string
@@ -375,7 +375,7 @@ export class Uppy {
 
   removePlugin(instance: UIPlugin | BasePlugin): void
 
-  close(options: CancelOptions): void
+  close(options?: CancelOptions): void
 
   logout(): void
 

